src/pyproject.toml.m4

```diff
@@ -1,24 +1,156 @@
+include(`sage_spkg_versions_toml.m4')dnl' -*- conf-toml -*-
 [build-system]
 # Minimum requirements for the build system to execute.
 requires = [
     "sage_setup[autogen]",
     # Some version of sage-conf is required.
     # Note that PEP517/518 have no notion of optional sage_spkg dependencies:
     # https://github.com/pypa/pip/issues/6144
-     esyscmd(`sage-get-system-packages install-requires-toml \
-        sage_conf      \
-        setuptools_wheel \
-        wheel          \
-        sage_setup     \
-        cypari         \
-        cysignals      \
-        cython         \
-        gmpy2          \
-        jinja2         \
-        jupyter_core   \
-        numpy          \
-        pkgconfig      \
-        pplpy          \
-        memory_allocator \
-                    ')]
+    SPKG_INSTALL_REQUIRES_sage_conf
+    SPKG_INSTALL_REQUIRES_setuptools
+    SPKG_INSTALL_REQUIRES_wheel
+    SPKG_INSTALL_REQUIRES_sage_setup
+    SPKG_INSTALL_REQUIRES_cypari
+    SPKG_INSTALL_REQUIRES_cysignals
+    SPKG_INSTALL_REQUIRES_cython
+    SPKG_INSTALL_REQUIRES_gmpy2
+    SPKG_INSTALL_REQUIRES_jupyter_core
+    SPKG_INSTALL_REQUIRES_memory_allocator
+    SPKG_INSTALL_REQUIRES_numpy
+    SPKG_INSTALL_REQUIRES_pkgconfig
+    SPKG_INSTALL_REQUIRES_pplpy
+]
 build-backend = "setuptools.build_meta"
+
+[project]
+name = "sagemath-standard"
+description = "Sage: Open Source Mathematics Software: Standard Python Library"
+dependencies = [
+    SPKG_INSTALL_REQUIRES_sage_conf
```

Comment:
I think you can simply use dynamic metadata and
```
dependencies = {file = ["build/pkgs/whatever/install-requires.txt", "build/pkgs/whatever2/install-requires.txt", ...]}
```

Reply:
Thanks, but that would not be an improvement.
It would make the sdist of this distribution depend on the monorepo / the Sage distribution.

Reply:
setuptools automatically includes the linked files (and only these) in the sdist.

Reply:
These file paths `build/pkgs/...` do not exist in this directory.

Reply:
In any case, changes to this design are outside of the scope of this PR. This design is already in place in
- https://github.com/sagemath/sage/blob/develop/pkgs/sagemath-bliss/pyproject.toml.m4
- https://github.com/sagemath/sage/blob/develop/pkgs/sagemath-categories/pyproject.toml.m4
- https://github.com/sagemath/sage/blob/develop/pkgs/sagemath-coxeter3/pyproject.toml.m4
- https://github.com/sagemath/sage/blob/develop/pkgs/sagemath-environment/pyproject.toml.m4
- https://github.com/sagemath/sage/blob/develop/pkgs/sagemath-mcqd/pyproject.toml.m4
- https://github.com/sagemath/sage/blob/develop/pkgs/sagemath-meataxe/pyproject.toml.m4
- https://github.com/sagemath/sage/blob/develop/pkgs/sagemath-objects/pyproject.toml.m4
- https://github.com/sagemath/sage/blob/develop/pkgs/sagemath-repl/pyproject.toml.m4
- https://github.com/sagemath/sage/blob/develop/pkgs/sagemath-sirocco/pyproject.toml.m4
- https://github.com/sagemath/sage/blob/develop/pkgs/sagemath-tdlib/pyproject.toml.m4